portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java

```diff
@@ -0,0 +1,58 @@
+package org.dataportabilityproject.types.transfer.models.calendar;
+
+// This interface could be used by other models.  If so, we should move it out and make it more
+// broadly useful.
+interface Property<T> {
+  T getValue();
+}
+
+class RuleDay implements Property<RecurringEvent.Day> {
+  private RecurringEvent.Day day;
+
+  RuleDay(RecurringEvent.Day day) {
+    this.day = day;
+  }
+
+  @Override
+  public RecurringEvent.Day getValue() {
+    return day;
+  }
+}
+
+class RuleNumber implements Property<Integer> {
+  private Integer number;
+
+  RuleNumber(Integer number) {
+    this.number = number;
+  }
+
+  @Override
+  public Integer getValue() {
+    return number;
+  }
+}
+
+class RuleByDay implements Property<ByDay> {
+  private ByDay byDay;
+
+  RuleByDay(ByDay byDay) {
+    this.byDay = byDay;
+  }
+
+  @Override
+  public ByDay getValue() {
+    return byDay;
+  }
+}
+
+class ByDay {
+  private boolean isNegative;
+  private RuleNumber number;
+  private RuleDay day;
+
+  ByDay(boolean isNegative, RuleNumber number, RuleDay day) {
+    this.isNegative = isNegative;
+    this.number = number;
+    this.day = day;
+  }
+}
```

Comment:
new line at the end (IntelliJ has this in settings on save)

Reply:
Done.  (Did not have that setting on my Mac, thanks for reminding me!)